lasagne/layers/dnn.py

```diff
@@ -10,16 +9,32 @@
 from .normalization import BatchNormLayer
 from ..utils import as_tuple
 
-if not theano.sandbox.cuda.cuda_enabled:
+try:
+    from theano import gpuarray as gpu
+except ImportError:
+    from theano.sandbox import gpuarray as gpu
+gpu_enabled = gpu.pygpu_activated
+dnn_enabled = gpu.dnn.dnn_present
+if not gpu_enabled:
+    try:
+        from theano.sandbox import cuda as gpu
```

Comment:
As per @lamblin's comment, we need to explicitly import dnn for the old backend. The simplest fix would be to add a line here (before the `except`): `import theano.sandbox.cuda.dnn`. This will also make dnn available as `gpu.dnn`.

A cleaner version would be to assign `dnn = gpu.dnn` above, after `dnn_enabled = ...`, and add `from theano.sandbox.cuda import dnn` here, and change the `if/else` construct below, but I'd be fine with the simpler solution.